From 80d357ea1d9b373ad6fe7ffdab1b10ff0765fcc4 Mon Sep 17 00:00:00 2001
From: danushree05 <danushree062005@gmail.com>
Date: Wed, 10 Jul 2024 22:59:03 +0530
Subject: [PATCH 1/4] update getDataPoint

---
 client3.py | 22 +---------------------
 1 file changed, 1 insertion(+), 21 deletions(-)

diff --git a/client3.py b/client3.py
index 8504d78..818d23f 100644
--- a/client3.py
+++ b/client3.py
@@ -35,24 +35,14 @@ def getDataPoint(quote):
     stock = quote['stock']
     bid_price = float(quote['top_bid']['price'])
     ask_price = float(quote['top_ask']['price'])
-<<<<<<< HEAD
-    price =(bid_price+ask_price)/2
-=======
-    price = bid_price
->>>>>>> 2c8cab305879a866937dfab235381b47662635ea
+    price = (bid_price+ask_price)/2
     return stock, bid_price, ask_price, price
 
 
 def getRatio(price_a, price_b):
     """ Get ratio of price_a and price_b """
     """ ------------- Update this function ------------- """
-<<<<<<< HEAD
-    if(price_b==0):
-        return
-    return price_a/price_b
-=======
     return 1
->>>>>>> 2c8cab305879a866937dfab235381b47662635ea
 
 
 # Main
@@ -62,18 +52,8 @@ if __name__ == "__main__":
         quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
         """ ----------- Update to get the ratio --------------- """
-<<<<<<< HEAD
-        prices={}
-        for quote in quotes:
-            stock, bid_price, ask_price, price = getDataPoint(quote)
-            prices[stock] = price
-            print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
-
-        print("Ratio %s" % getRatio(prices["ABC"], prices["DEF"]))
-=======
         for quote in quotes:
             stock, bid_price, ask_price, price = getDataPoint(quote)
             print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
         print("Ratio %s" % getRatio(price, price))
->>>>>>> 2c8cab305879a866937dfab235381b47662635ea
-- 
2.39.1.windows.1


From 4a4a7bde1421716e8c09cadd6410e8dee68501b2 Mon Sep 17 00:00:00 2001
From: danushree05 <danushree062005@gmail.com>
Date: Wed, 10 Jul 2024 22:59:58 +0530
Subject: [PATCH 2/4] update getRatio

---
 client3.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/client3.py b/client3.py
index 818d23f..de8d053 100644
--- a/client3.py
+++ b/client3.py
@@ -42,7 +42,9 @@ def getDataPoint(quote):
 def getRatio(price_a, price_b):
     """ Get ratio of price_a and price_b """
     """ ------------- Update this function ------------- """
-    return 1
+    if(price_b==0):
+        return
+    return price_a/price_b
 
 
 # Main
-- 
2.39.1.windows.1


From 11b8c933c0973b1c521e6768ddac3cf272ffe97b Mon Sep 17 00:00:00 2001
From: danushree05 <danushree062005@gmail.com>
Date: Wed, 10 Jul 2024 23:00:59 +0530
Subject: [PATCH 3/4] update main

---
 client3.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/client3.py b/client3.py
index de8d053..1da3ea4 100644
--- a/client3.py
+++ b/client3.py
@@ -54,8 +54,10 @@ if __name__ == "__main__":
         quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
         """ ----------- Update to get the ratio --------------- """
+        prices={}
         for quote in quotes:
             stock, bid_price, ask_price, price = getDataPoint(quote)
+            prices[stock]=price
             print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-        print("Ratio %s" % getRatio(price, price))
+        print("Ratio %s" % getRatio(prices["ABC"], prices["DEF"]))
-- 
2.39.1.windows.1


From 04af6052b1b82c8b58aaf92454b145f1c4bfda80 Mon Sep 17 00:00:00 2001
From: danushree05 <danushree062005@gmail.com>
Date: Wed, 10 Jul 2024 23:03:16 +0530
Subject: [PATCH 4/4] Tested

---
 client_test.py | 31 +++++++++++++++++++++++++++++--
 1 file changed, 29 insertions(+), 2 deletions(-)

diff --git a/client_test.py b/client_test.py
index af2bf26..e230000 100644
--- a/client_test.py
+++ b/client_test.py
@@ -1,5 +1,5 @@
 import unittest
-from client3 import getDataPoint
+from client3 import getDataPoint,getRatio
 
 class ClientTest(unittest.TestCase):
   def test_getDataPoint_calculatePrice(self):
@@ -7,6 +7,12 @@ class ClientTest(unittest.TestCase):
       {'top_ask': {'price': 121.2, 'size': 36}, 'timestamp': '2019-02-11 22:06:30.572453', 'top_bid': {'price': 120.48, 'size': 109}, 'id': '0.109974697771', 'stock': 'ABC'},
       {'top_ask': {'price': 121.68, 'size': 4}, 'timestamp': '2019-02-11 22:06:30.572453', 'top_bid': {'price': 117.87, 'size': 81}, 'id': '0.109974697771', 'stock': 'DEF'}
     ]
+    expected = [
+            ('ABC', 120.48, 121.2, (120.48 + 121.2) / 2),
+            ('DEF', 117.87, 121.68, (117.87 + 121.68) / 2)
+        ]
+    for i, quote in enumerate(quotes):
+            self.assertEqual(getDataPoint(quote), expected[i])
     """ ------------ Add the assertion below ------------ """
 
   def test_getDataPoint_calculatePriceBidGreaterThanAsk(self):
@@ -15,10 +21,31 @@ class ClientTest(unittest.TestCase):
       {'top_ask': {'price': 121.68, 'size': 4}, 'timestamp': '2019-02-11 22:06:30.572453', 'top_bid': {'price': 117.87, 'size': 81}, 'id': '0.109974697771', 'stock': 'DEF'}
     ]
     """ ------------ Add the assertion below ------------ """
+    expected = [
+            ('ABC', 120.48, 119.2, (120.48 + 119.2) / 2),
+            ('DEF', 117.87, 121.68, (117.87 + 121.68) / 2)
+        ]
+    for i, quote in enumerate(quotes):
+            self.assertEqual(getDataPoint(quote), expected[i])
 
-
+    
   """ ------------ Add more unit tests ------------ """
+  def test_getRatio_calculateRatio(self):
+        price_a = 121.68
+        price_b = 117.87
+        expected = price_a / price_b
+        self.assertEqual(getRatio(price_a, price_b), expected)
+
+  def test_getRatio_priceBZero(self):
+        price_a = 121.68
+        price_b = 0
+        self.assertIsNone(getRatio(price_a, price_b))
 
+  def test_getRatio_priceAZero(self):
+        price_a = 0
+        price_b = 117.87
+        expected = 0.0
+        self.assertEqual(getRatio(price_a, price_b), expected)
 
 
 if __name__ == '__main__':
-- 
2.39.1.windows.1

